### DISCUSSION TITLE
basicvsr calculate y-channel psnr

### DISCUSSION THREAD

**Discussion by youyoubud** [open-mmlab/mmediting#1723] [Datasets] [answered]
In basicvsr_vimeo90k_bi.py: 
train_pipeline loads image in RGB channel order, and calculates y-channel psnr.
but  when computing psnr for 'Y', the images are assumed to be in BGR order.
why?🙏

`def psnr(img1, img2, crop_border=0, input_order='HWC', convert_to=None):
    """Calculate PSNR (Peak Signal-to-Noise Ratio).

    Ref: https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio

    Args:
        img1 (ndarray): Images with range [0, 255].
        img2 (ndarray): Images with range [0, 255].
        crop_border (int): Cropped pixels in each edges of an image. These
            pixels are not involved in the PSNR calculation. Default: 0.
        input_order (str): Whether the input order is 'HWC' or 'CHW'.
            Default: 'HWC'.
        convert_to (str): Whether to convert the images to other color models.
            If None, the images are not altered. When computing for 'Y',
            the images are assumed to be in BGR order. Options are 'Y' and
            None. Default: None.

    Returns:
        float: psnr result.
    """`

**Comment by Z-Fran** [comment, marked as answer]
Hi, you can view [these codes](https://github.com/open-mmlab/mmediting/blob/master/mmedit/models/restorers/basicvsr.py#L116-L149). Before calling `psnr()`, we will use `tensor2img()` to convert the image to `BGR`. And we recommend you use [new version MMEditing](https://github.com/open-mmlab/mmediting/tree/1.x)

**Reply by youyoubud** [reply]
thank you! 🌹
